legal-api/src/legal_api/services/filings/validations/continuation_out.py

```diff
@@ -0,0 +1,90 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Validation for the Continuation Out filing."""
+from http import HTTPStatus
+from typing import Dict, Final, Optional
+
+import pycountry
+from flask_babel import _ as babel  # noqa: N813, I004, I001; importing camelcase '_' as a name
+# noqa: I003
+from legal_api.errors import Error
+from legal_api.models import Business
+from legal_api.services.filings.validations.common_validations import validate_court_order
+from legal_api.services.utils import get_date
+from legal_api.utils.legislation_datetime import LegislationDatetime
+# noqa: I003;
+
+
+def validate(business: Business, filing: Dict) -> Optional[Error]:
+    """Validate the Continuation Out filing."""
+    if not business or not filing:
+        return Error(HTTPStatus.BAD_REQUEST, [{'error': babel('A valid business and filing are required.')}])
+
+    msg = []
+    filing_type = 'continuationOut'
+
+    msg.extend(validate_continuation_out_date(business, filing, filing_type))
+    msg.extend(validate_foreign_jurisdiction(filing, filing_type))
+
+    if court_order := filing.get('filing', {}).get(filing_type, {}).get('courtOrder', None):
+        court_order_path: Final = f'/filing/{filing_type}/courtOrder'
+        err = validate_court_order(court_order_path, court_order)
+        if err:
+            msg.extend(err)
+
+    if msg:
+        return Error(HTTPStatus.BAD_REQUEST, msg)
+    return None
+
+
+def validate_continuation_out_date(business: Business, filing: Dict, filing_type: str) -> list:
+    """Validate continuation out date."""
+    msg = []
+    continuation_out_date_path = f'/filing/{filing_type}/continuationOutDate'
+    continuation_out_date = get_date(filing, continuation_out_date_path)
+
+    now = LegislationDatetime.now().date()
+    if continuation_out_date > now:
+        msg.append({'error': 'Continuation out date must be today or past.',
+                    'path': continuation_out_date_path})
+
+    if business.cco_expiry_date:
+        cco_expiry_date = LegislationDatetime.as_legislation_timezone(business.cco_expiry_date).date()
+        if continuation_out_date > cco_expiry_date:
+            msg.append({'error': 'Consent continuation of interest has expired.',
+                        'path': continuation_out_date_path})
+    else:
+        msg.append({'error': 'Did not find an active Consent continuation out for this business.',
+                    'path': continuation_out_date_path})
+
+    return msg
+
+
+def validate_foreign_jurisdiction(filing: Dict, filing_type: str) -> list:
+    """Validate foreign jurisdiction."""
+    msg = []
+    foreign_jurisdiction_path = f'/filing/{filing_type}/foreignJurisdiction'
+
+    foreign_jurisdiction = filing['filing'][filing_type]['foreignJurisdiction']
+    country = pycountry.countries.search_fuzzy(foreign_jurisdiction['country'])[0].alpha_2
+    region = foreign_jurisdiction.get('region')
```

Comment:
i think we'll need to deal with `undefined` or `null` scenarios for `region`

Reply:
you mean to make sure its `undefined` or `null` when country is other than `US ot CA`?

Reply:
no, for `CA` and `US` scenarios.  

I think `if region == 'BC'` will be an issue. 

 Not sure if `elif not (region == 'FEDERAL' or pycountry.subdivisions.get(code=f'{country}-{region}')):` will error out.

Reply:
> I think `if region == 'BC'` will be an issue.

There are lots of extra checks we could do but I agree this is a good one. (Even though our UI won't allow this region, the API is used by external users and should protect itself.)

Reply:
actually, I think your code works.  Resolving.